engine-precompiles/src/secp256k1.rs

```diff
@@ -19,7 +20,9 @@ mod consts {
 // Quite a few library methods rely on this and that should be changed. This
 // should only be for precompiles.
 pub fn ecrecover(hash: H256, signature: &[u8]) -> Result<Address, ExitError> {
-    assert_eq!(signature.len(), 65);
+    if signature.len() != consts::SIGNATURE_LENGTH {
+        return Err(ExitError::Other(Borrowed("INVALID_SIGNATURE_LENGTH")));
+    }
```

Comment:
Instead of returning an error, what if we changed the type signature to `signature: &[u8; consts::SIGNATURE_LENGTH]`? All our internal usage of this function is already doing that, so no other changes should be needed.

Yes, this would technically be a breaking change to a public API, but I think it's fine because I doubt anyone is using this function in their project and if they are then it should be easy enough for them to fix it at their call-site.